tools/osbuilder/rootfs-builder/init_trusted_storage.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+#
+# Copyright (c) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+[ -n "${BASH_VERSION:-}" ] && set -o errtrace
+[ -n "${DEBUG:-}" ] && set -o xtrace
+
+device_num=${1:-}
+if [ -z "$device_num" ]; then
+        echo "ERROR: invalid arguments" >&2
+        exit 1
+fi
+
+enable_data_integrity="true"
+if [ -n "${2-}" ]; then
+        enable_data_integrity="$2"
+fi
+
+device_name=$(cat /sys/dev/block/${device_num}/uevent|grep DEVNAME|cut -d "=" -f2)
+device_path="/dev/$device_name"
+if [[ -n "$device_name" && -b "$device_path" ]]; then
+	dd if=/dev/urandom of=/run/cc_storage.key bs=1 count=4096
+
+	if [ "$enable_data_integrity" == "false" ]; then
+		echo "YES" | cryptsetup luksFormat --type luks2 "$device_path" --sector-size 4096 \
+			--cipher aes-xts-plain64 /run/cc_storage.key
+	else
+		echo "YES" | cryptsetup luksFormat --type luks2 "$device_path" --sector-size 4096 \
+			 --cipher aes-xts-plain64 --integrity hmac-sha256 /run/cc_storage.key
+	fi
+
+	cryptsetup luksOpen -d /run/cc_storage.key "$device_path" encrypted_disk
```

Comment:
In case of failure of cryptsetup, we get no message. The script will just exit, but it will not be very helpful.

Consider adding an error handler to display an error similar to the other ones?

```
error_handler() {
    echo "ERROR: $0:$2: Unexpected error $1 from cryptsetup, exiting"
    exit $1
}
trap 'error_handler $? $LINENO' ERR
```